src/xrdsvc/HttpWorkerCzarModule.cc

```diff
@@ -0,0 +1,213 @@
+        LOGS(_log, LOG_LVL_TRACE,
+             __func__ << " uj qid=" << ujQueryId << " ujid=" << ujId << " czid=" << ujCzarId);
+
+        auto ujData = wbase::UberJobData::create(ujId, czarName, czarId, czarHostName, czarPort, ujQueryId,
+                                                 targetWorkerId, foreman(), authKey());
+
+        // Find the entry for this queryId, creat a new one if needed.
+        wbase::UserQueryInfo::Ptr userQueryInfo = wbase::UserQueryInfo::uqMapInsert(ujQueryId);
+        userQueryInfo->addUberJob(ujData);
+
+        auto channelShared =
+                wbase::FileChannelShared::create(ujData, czarId, czarHostName, czarPort, targetWorkerId);
+        ujData->setFileChannelShared(channelShared);
+
+        // TODO:UJ These items should be stored higher in the message structure as they get
+        //   duplicated and should always be the same within an UberJob.
+        QueryId jdQueryId = 0;
+        proto::ScanInfo scanInfo;
+        bool scanInfoSet = false;
+        bool jdScanInteractive = false;
+        int jdMaxTableSize = 0;
+
+        for (auto const& job : ujJobs) {
+            json const& jsJobDesc = job["jobdesc"];
+            http::RequestBody rbJobDesc(jsJobDesc);
+            // See qproc::TaskMsgFactory::makeMsgJson for message construction.
+            auto const jdCzarId = rbJobDesc.required<qmeta::CzarId>("czarId");
+            jdQueryId = rbJobDesc.required<QueryId>("queryId");
+            auto const jdJobId = rbJobDesc.required<int>("jobId");
+            auto const jdAttemptCount = rbJobDesc.required<int>("attemptCount");
+            auto const jdQuerySpecDb = rbJobDesc.required<string>("querySpecDb");
+            auto const jdScanPriority = rbJobDesc.required<int>("scanPriority");
+            jdScanInteractive = rbJobDesc.required<bool>("scanInteractive");
+            jdMaxTableSize = rbJobDesc.required<int>("maxTableSize");
+            auto const jdChunkId = rbJobDesc.required<int>("chunkId");
+            LOGS(_log, LOG_LVL_TRACE,
+                 __func__ << " jd cid=" << jdCzarId << " jdQId=" << jdQueryId << " jdJobId=" << jdJobId
+                          << " jdAtt=" << jdAttemptCount << " jdQDb=" << jdQuerySpecDb
+                          << " jdScanPri=" << jdScanPriority << " interactive=" << jdScanInteractive
+                          << " maxTblSz=" << jdMaxTableSize << " chunkId=" << jdChunkId);
+
+            auto const jdChunkScanTables = rbJobDesc.required<json>("chunkScanTables");
+            if (!scanInfoSet) {
+                for (auto const& tbl : jdChunkScanTables) {
+                    http::RequestBody rbTbl(tbl);
+                    auto const& chunkScanDb = rbTbl.required<string>("db");
+                    auto const& lockInMemory = rbTbl.required<bool>("lockInMemory");
+                    auto const& chunkScanTable = rbTbl.required<string>("table");
+                    auto const& tblScanRating = rbTbl.required<int>("tblScanRating");
```

Comment:
You don't really gain anything by having references to the primitive data types. The references are still going to be converted to values a few lines below where you are calling:
```
scanInfo.infoTables.emplace_back(chunkScanDb, chunkScanTable, lockInMemory,
                                                     tblScanRating);
```
The only variable that may make a sense to pass by reference would be a string stored in `chunkScanTable`.